### ISSUE TITLE
`yarn rw g scaffold` creates delete mutation with wrong `non-nullable` return type

### ISSUE THREAD

**Issue by zygopleural** [redwoodjs/redwood#3902]
# Expected behaviour

`yarn rw g scaffold <model>` should generate a `./api/src/graphql/models.sdl.ts` with a mutation:
```typescript
type Mutation {
  ...
  deleteModel(id: Int!): Model @requireAuth
}
```

i.e. It should return a nullable model instance (since the model would have been deleted).

**Is this expected? E.g. is this meant to be returning a snapshot of the model instance before deletion?**

# What currently happens

The props are created as:
```typescript
type Mutation {
  ...
  deleteModel(id: Int!): Model! @requireAuth 👈 problem here
}
```

# Steps to reproduce

### 1. Create new app
`yarn create redwood-app --typescript test`

### 2. Add model to `./api/db/schema.prisma`
```prisma
model Example {
  id        Int    @id @default(autoincrement())
  something String
}
```

### 3. Migrate
`yarn rw prisma migrate dev`

### 4. Scaffold
`yarn rw g scaffold example`

### 5. Check
Take a look at `L23` in `./api/src/graphql/examples.sdl.ts`

**Comment by jtoar** [comment]
Hey @zygopleural, thanks for the detailed issue! 

> Is this expected? E.g. is this meant to be returning a snapshot of the model instance before deletion?

TL;DR, yes—the resolver generated by scaffolds for the `deleteModel` mutation uses the prisma client's `delete` function. So it looks something like

```js
export const deleteModel = ({ id }) => {
  return db.model.delete({
    where: { id },
  })
}
``` 

This returns the deleted record[^1], so the GraphQL typing isn't a bug per se, but if you'd like us to reconsider the typing we can.

[^1]: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#return-type-6

**Comment by zygopleural** [comment]
Hi @jtoar 👋🏼

Thanks for the quick response.

Yeah that makes perfect sense, thanks for the link to Prisma docs.

I'll close this as it is not an issue 👌🏼
